_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_splitter.cpp

```diff
@@ -0,0 +1,259 @@
+            // An exception from this is 'MFX_BITSTREAM_COMPLETE_FRAME' case.
+            // These two bytes are to prevent start code chopping between two separete FindRawUnit() calls.
+            uint32_t numBytesToLeave = !((m_source->GetFlags() & UMC::MediaData::FLAG_VIDEO_DATA_NOT_FULL_FRAME)) ? 0 : (prefix_size - 1);
+            readSize = end - begin - numBytesToLeave;
+            m_cache.insert(m_cache.end(), naluTmp.begin, end - numBytesToLeave); // But save to the cache only since start code
+            m_pts = in->GetTime();
+        }
+        else // found begin and end
+        {
+            readSize = naluTmp.end - begin; // Up to the end of the unit
+            unit = naluTmp;
+            unit.pts = in->GetTime();
+        }
+
+        in->MoveDataPointer(readSize);
+
+        return unit;
+    }
+
+    RawUnit RawHeaderIterator::OnEOS()
+    {
+        RawUnit nalu;
+        if (m_needCleanCache)
+        {
+            m_cache.clear();
+            m_needCleanCache = false;
+            m_pts = -1;
+        }
+
+        if (m_cache.size())
+        {
+            nalu = { m_cache.data(), m_cache.data() + m_cache.size(), m_cache[prefix_size], m_pts } ; // Construct unit from cached data
+            m_needCleanCache = true; // Need to clean up the cache on next iteration
+        }
+
+        return nalu;
+    }
+
+    uint8_t * RawHeaderIterator::FindStartCode(uint8_t * begin, uint8_t * end)
+    {
+        for (; begin < end - 3; ++begin)
+        {
+            if (begin[0] == 0 && begin[1] == 0 && begin[2] == 1)
+            {
+                return begin;
+            }
+        }
+        return nullptr;
+    }
+
+    RawUnit RawHeaderIterator::FindRawUnit(mfxU8 * begin, mfxU8 * end)
+    {
+        for (; begin < end - 4; ++begin)
+        {
+            if (begin[0] == 0 && begin[1] == 0 && begin[2] == 1)
+            {
+                mfxU8 type = begin[3];
+
+                for (mfxU8 * next = begin + 4; next < end - 3; ++next)
+                {
+                    if (next[0] == 0 && next[1] == 0 && next[2] == 1)
+                    {
+                        return RawUnit(begin, next, type, -1);
+                    }
+                }
+
+                return RawUnit(begin, nullptr, type, -1);
+            }
+        }
+
+        return RawUnit();
+    }
+
+    RawUnit Splitter::GetUnits(UMC::MediaData* source)
+    {
+        if (m_cleanUpCache)
+        {
+            m_buff.clear();
+            m_cleanUpCache = false;
+        }
+
+        m_it.LoadData(source); // Find new unit
+
+        // for loop is to handle cases like: seq header, seq header, seq extension
+        for (;;)
+        {
+            if (m_buff.size() && (m_it != RawHeaderIterator()))
+            {
+                size_t size = m_it->end - m_it->begin;
+                MPEG2_UNIT_TYPE_EXT type_ext = (MPEG2_UNIT_TYPE_EXT) ((m_it->type == EXTENSION && size >= 5) ? (m_it->begin[prefix_size + 1] >> 4) : 0 );
```

Comment:
What is 5? Is it length of extension header? Need to either add self-descriptive constant or comment at least.
BTW, I'd rather make [size] to be method of _RawHeaderIterator_ to avoid mixing low level pointers logic and high level units handling.

The same concerns about _m_it->begin[prefix_size + 1] >> 4_

Reply:
>>What is 5
Because of m_it->begin[prefix_size + 1] >> 4. In the same line. Ok, I'll add a comment. 

>>BTW, I'd rather make [size] to be method of RawHeaderIterator to avoid mixing low level pointers logic and high level units handling.
It' will bring too many changes without actual benefits.  I'd like to avoid it in current PR.